game-app/game-core/src/main/java/games/strategy/triplea/attachments/PoliticalActionAttachment.java

```diff
@@ -152,22 +150,62 @@ public List<RelationshipChange> getRelationshipChanges() {
+    try {
+      return new RelationshipChange(
+          getPlayerByName(tokens[0])
+              .orElseThrow(
+                  () ->
+                      new GameParseException(
+                          MessageFormat.format(
+                              "Invalid relationshipChange declaration: {0} \n first player: {1} unknown{2}",
+                              encodedRelationshipChange, tokens[0], thisErrorMsg()))),
+          getPlayerByName(tokens[1])
+              .orElseThrow(
+                  () ->
+                      new GameParseException(
+                          MessageFormat.format(
+                              "Invalid relationshipChange declaration: {0} \n second player: {1} unknown{2}",
+                              encodedRelationshipChange, tokens[1], thisErrorMsg()))),
+          getData().getRelationshipTypeList().getRelationshipType(tokens[2]));
+    } catch (GameParseException gameParseException) {
+      // @TODO: relationshipChanges should not be a list of strings that have to be parsed all the
+      // time;
+      // separate object needed in the future to ensure parsing is done only once
+      throw new IllegalStateException(gameParseException);
+    }
   }
 
   /** Returns a set of all other players involved in this PoliticalAction. */
   public Set<GamePlayer> getOtherPlayers() {
     final Set<GamePlayer> otherPlayers = new LinkedHashSet<>();
-    for (final String relationshipChange : getRelationshipChange()) {
-      final String[] s = splitOnColon(relationshipChange);
-      otherPlayers.add(getData().getPlayerList().getPlayerId(s[0]));
-      otherPlayers.add(getData().getPlayerList().getPlayerId(s[1]));
+    for (final String currentRelationshipChange : getRelationshipChange()) {
+      final String[] s = splitOnColon(currentRelationshipChange);
+      try {
+        otherPlayers.add(
+            getPlayerByName(s[0])
+                .orElseThrow(
+                    () ->
+                        new GameParseException(
+                            MessageFormat.format(
+                                "Invalid otherPlayers for relationshipChange: {0} \n first player: {1} unknown{2}",
+                                currentRelationshipChange, s[0], thisErrorMsg()))));
+        otherPlayers.add(
+            getPlayerByName(s[1])
+                .orElseThrow(
+                    () ->
+                        new GameParseException(
+                            MessageFormat.format(
+                                "Invalid otherPlayers for relationshipChange: {0} \n second player: {1} unknown{2}",
+                                currentRelationshipChange, s[1], thisErrorMsg()))));
+      } catch (GameParseException gameParseException) {
+        // @TODO: relationshipChange should not be a string that has to be parsed all the time;
+        // once separate object re-parsing here is not needed anymore
+        throw new IllegalStateException(gameParseException);
```

Comment:
ditto on avoiding 'try/catch' and just throw illegalStateException to begin with